numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -3541,3 +3530,26 @@ def inspect_obj_content(objpath: str):
             code_types.add(match.group(1))
 
     return code_types
+
+
+def _stream_handle(stream):
+    """
+    Obtain the appropriate handle for various types of
+    acceptable stream objects. Acceptable types are
+    int (0 for default stream), Stream, ExperimentalStream
+    """
+
+    if stream == 0:
+        return stream
+    elif hasattr(stream, "__cuda_stream__"):
+        _, ptr = stream.__cuda_stream__()
+        if isinstance(ptr, binding.CUstream):
+            return get_cuda_native_handle(ptr)
```

Comment:
this seems like a bug in cuda.core, I vaguely remember we discussed this, is there an issue created already?

Reply:
FWIW, we've fixed this in v0.3.1 (as part of https://github.com/NVIDIA/cuda-python/pull/709).